jolokia-cassandra-stream/getmetrics_jolokia.py

```diff
@@ -0,0 +1,222 @@
+#!/bin/python
+import subprocess
+import json
+import os
+import time
+from optparse import OptionParser
+import csv
+from ConfigParser import SafeConfigParser
+import requests
+
+COMMA_DELIMITER = ","
+NEWLINE_DELIMITER = "\n"
+os_metrics = {'Memory': {'NonHeapMemoryUsage': ['max', 'committed', 'init', 'used'],
+                         'HeapMemoryUsage': ['max', 'committed', 'init', 'used']},
+              'OperatingSystem': ['ProcessCpuLoad', 'SystemCpuLoad', 'MaxFileDescriptorCount',
+                                  'OpenFileDescriptorCount'], 'Threading': ['ThreadCount']}
+cassandra_metrics = {
+#    'org.apache.cassandra.metrics:name=TotalLatency,scope=Read,type=ClientRequest': ['Count'],
+#    'org.apache.cassandra.metrics:name=TotalLatency,scope=Write,type=ClientRequest': ['Count'],
+    'org.apache.cassandra.metrics:name=Latency,scope=Read,type=ClientRequest': ['OneMinuteRate'],
+    'org.apache.cassandra.metrics:name=Latency,scope=Write,type=ClientRequest': ['OneMinuteRate'],
+    'org.apache.cassandra.metrics:name=Timeouts,scope=Read,type=ClientRequest': ['Count'],
+    'org.apache.cassandra.metrics:name=Timeouts,scope=Write,type=ClientRequest': ['Count'],
+    'org.apache.cassandra.metrics:name=Unavailables,scope=Read,type=ClientRequest': ['Count'],
+    'org.apache.cassandra.metrics:name=Unavailables,scope=Write,type=ClientRequest': ['Count'],
+    'org.apache.cassandra.metrics:name=Failures,scope=Read,type=ClientRequest': ['Count'],
+    'org.apache.cassandra.metrics:name=Failures,scope=Write,type=ClientRequest': ['Count']
+}
+
+
+def get_name_from_id(mbean_id):
+    full_path = mbean_id.split(":")[1]
+    path_vals = full_path.split(",")
+    metric_name = ""
+    for entry in path_vals:
+        metric_name += entry.split("=")[1] + "_"
+    metric_name = metric_name[:-1]
+    return metric_name
+
+
+def getInstanceDetails(filename):
+    instances = []
+    with open(filename, 'rb') as csvFile:
+        instancereader = csv.reader(csvFile, delimiter=',')
+        for row in instancereader:
+            if len(row) == 0:
+                continue
+            instance = []
+            instance.append(row[0])
+            instance.append(row[1])
+            instances.append(instance)
+    return instances
+
+
+def getindex(col_name):
+    if col_name == "NonHeapMemoryUsage":
+        return 9001
+    elif col_name == "HeapMemoryUsage":
+        return 9002
+    elif col_name == "ProcessCpuLoad":
+        return 9003
+    elif col_name == "SystemCpuLoad":
+        return 9004
+    elif "ClientRequest" in col_name:
+        return 9200
+
+def get_agent_config_vars():
+    config_vars = {}
+    try:
+        if os.path.exists(os.path.join(homepath, "config.ini")):
+            parser = SafeConfigParser()
+            parser.read(os.path.join(homepath, "config.ini"))
+            #file_name = parser.get('InsightFinder', 'file_name')
```

Comment:
if you do not support file, please remove it in both code and config file rather than comment